core/components/tree/basenode.js

```diff
@@ -892,22 +728,13 @@ Blockly.tree.BaseNode.prototype.getRowElement = function() {
   return el ? /** @type {Element} */ (el.firstChild) : null;
 };
 
-/**
- * @return {Element} The expanded icon element.
- * @protected
- */
-Blockly.tree.BaseNode.prototype.getExpandIconElement = function() {
-  var el = this.getRowElement();
-  return el ? /** @type {Element} */ (el.firstChild) : null;
-};
-
 /**
  * @return {Element} The icon element.
  * @protected
  */
 Blockly.tree.BaseNode.prototype.getIconElement = function() {
   var el = this.getRowElement();
-  return el ? /** @type {Element} */ (el.childNodes[1]) : null;
+  return el ? /** @type {Element} */ (el.firstChild) : null;
```

Comment:
This changes the DOM structure of the toolbox which might break some legacy CSS out there that assumes the icon element to be the second element.

Reply:
Given that the icon element has both a class and an id, I think that any reasonable developer would use one of those rather than child order.